### PR TITLE
[fix][admin] Fix NPE in PulsarAdminBuilder when the service is not set

### LINKED ISSUE #14768: PulsarAdmin throws NPE when the service url is not set
**Describe the bug**

PulsarAdmin throws NPE when the service url is not set

**To Reproduce**
1. Create a PulsarAdminBuilder with service URL not set
2. It will throws NPE when build the pulsar admin

```
Caused by: java.lang.NullPointerException
	at org.apache.pulsar.client.admin.internal.PulsarAdminImpl.<init>(PulsarAdminImpl.java:191)
	at org.apache.pulsar.client.admin.internal.PulsarAdminBuilderImpl.build(PulsarAdminBuilderImpl.java:47)
	at org.apache.pulsar.sql.presto.PulsarConnectorConfig.getPulsarAdmin(PulsarConnectorConfig.java:436)
	... 51 more
```

**Expected behavior**
Throw IlleglaArgumentException instead of NPE.

### PR DESCRIPTION
**Pull request by RobertIndie**
Fixes #14768 

### Motivation

PulsarAdmin throws NPE when the service url is not set
```
Caused by: java.lang.NullPointerException
	at org.apache.pulsar.client.admin.internal.PulsarAdminImpl.<init>(PulsarAdminImpl.java:191)
	at org.apache.pulsar.client.admin.internal.PulsarAdminBuilderImpl.build(PulsarAdminBuilderImpl.java:47)
	at org.apache.pulsar.sql.presto.PulsarConnectorConfig.getPulsarAdmin(PulsarConnectorConfig.java:436)
	... 51 more
```

### Modifications

* Throw IlleglaArgumentException instead of NPE when the service URL is not set.

### Verifying this change

This change is already covered by existing tests, such as *testAdminBuilderWithServiceUrlNotSet*.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

- [x] `no-need-doc` 
  
This is a bug fix.

### CODE REVIEW

**Review by Shoothzj — APPROVED** [review]

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/PulsarAdminBuilderImpl.java [hunk 1]
```diff
@@ -44,6 +46,8 @@
 
     @Override
     public PulsarAdmin build() throws PulsarClientException {
+        checkArgument(StringUtils.isNotBlank(conf.getServiceUrl()), "Service URL needs to be specified");
```
Comment (eolivelli) [inline pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/PulsarAdminBuilderImpl.java:49]: Ehy mot oushing this check inside the constructor of PulsarAdminImpl?

**Review by eolivelli — CHANGES_REQUESTED** [review]

**Review by Jason918 — APPROVED** [review]
LGTM

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/PulsarAdminBuilderImpl.java (hunk 1 above)
Comment (RobertIndie) [reply pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/PulsarAdminBuilderImpl.java:49]: The `ClientBuilder` also does the same way. https://github.com/apache/pulsar/blob/5ebaafd4288c69d135b295ab563c8ad56affa15b/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java#L53

Comment (codelipenghui) [reply pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/PulsarAdminBuilderImpl.java:49]: +1 

It's better to move to the constructor of PulsarAdminImpl.
We can change the ClientBuilderImpl.java

Comment (RobertIndie) [reply pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/PulsarAdminBuilderImpl.java]: @codelipenghui @eolivelli  I have moved this argument check to the constructor. PTAL. Thanks.

**Review by codelipenghui — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
Lgtm

**Pull request merged**